ui/ducks/swaps/swaps.js

```diff
@@ -342,6 +409,45 @@ export const getApproveTxParams = (state) => {
   return { ...approvalNeeded, gasPrice, data };
 };
 
+export const getSmartTransactionsOptInStatus = (state) => {
+  return state.metamask.smartTransactionsState?.userOptIn;
+};
+
+export const getCurrentSmartTransactions = (state) => {
+  return state.metamask.smartTransactionsState?.smartTransactions?.[
+    getCurrentChainId(state)
+  ];
+};
+
+export const getPendingSmartTransactions = (state) => {
+  const currentSmartTransactions = getCurrentSmartTransactions(state);
+  if (!currentSmartTransactions || currentSmartTransactions.length === 0) {
+    return [];
+  }
+  return currentSmartTransactions.filter((stx) => stx.status === 'pending');
```

Comment:
Probably use our constant pattern for `'pending'`? Declare a constant in `shared/constants/transaction.js` and import it here